packages/graph/hash_graph/lib/graph/src/knowledge/entity/query.rs

```diff
@@ -13,20 +13,239 @@ use crate::{
+    ///
+    /// Deserializes from `["type", ...]` where `...` is a path to a field of an [`EntityType`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::{knowledge::EntityQueryPath, ontology::EntityTypeQueryPath};
+    /// let path = EntityQueryPath::deserialize(json!(["type", "baseUri"]))?;
+    /// assert_eq!(path, EntityQueryPath::Type(EntityTypeQueryPath::BaseUri));
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`Entity`]: crate::knowledge::Entity
+    /// [`EntityMetadata`]: crate::knowledge::EntityMetadata
+    /// [`EntityType`]: type_system::EntityType
     Type(EntityTypeQueryPath),
-    Properties(Option<Cow<'q, str>>),
+    /// Represents an [`Entity`] linking to this [`Entity`].
+    ///
+    /// Deserializes from `["incomingLinks", ...]` where `...` is the path of the source
+    /// [`Entity`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["incomingLinks", "uuid"]))?;
+    /// assert_eq!(
+    ///     path,
+    ///     EntityQueryPath::IncomingLinks(Box::new(EntityQueryPath::Uuid))
+    /// );
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`Entity`]: crate::knowledge::Entity
     IncomingLinks(Box<Self>),
+    /// Represents an [`Entity`] linked from this [`Entity`].
+    ///
+    /// Deserializes from `["outgoingLinks", ...]` where `...` is the path of the target
+    /// [`Entity`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["outgoingLinks", "uuid"]))?;
+    /// assert_eq!(
+    ///     path,
+    ///     EntityQueryPath::OutgoingLinks(Box::new(EntityQueryPath::Uuid))
+    /// );
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`Entity`]: crate::knowledge::Entity
     OutgoingLinks(Box<Self>),
+    /// Corresponds to [`LinkEntityMetadata::left_entity_id()`].
+    ///
+    /// Deserializes from `["leftEntity", ...]` where `...` is the path of the left [`Entity`].
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["leftEntity", "uuid"]))?;
+    /// assert_eq!(
+    ///     path,
+    ///     EntityQueryPath::LeftEntity(Some(Box::new(EntityQueryPath::Uuid)))
+    /// );
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`Entity`]: crate::knowledge::Entity
+    /// [`LinkEntityMetadata::left_entity_id()`]: crate::knowledge::LinkEntityMetadata::left_entity_id
     LeftEntity(Option<Box<Self>>),
+    /// Corresponds to [`LinkEntityMetadata::right_entity_id()`].
```

Comment:
Same here and for the next. I think we can keep the wording for left/right order though :)

Also could you just remind me what the None case did here?